migration/idempotent/001-base.sql

```diff
@@ -3375,24 +3375,29 @@ GRANT EXECUTE ON FUNCTION _prom_catalog.insert_metric_row(TEXT, TIMESTAMPTZ[], D
 -- Suppresses corresponding DDL logging, otherwise PG log may get unnecessarily verbose.
 -- Temporary table is created using supplied table and schema as prototype.
 -- Returns temporary table name
-CREATE OR REPLACE FUNCTION _prom_catalog.create_ingest_temp_table(table_name TEXT, schema_name TEXT)
+CREATE OR REPLACE FUNCTION _prom_catalog.create_ingest_temp_table(table_name TEXT, schema_name TEXT, table_prefix TEXT)
     RETURNS TEXT
     SECURITY DEFINER
     VOLATILE
     SET search_path = pg_catalog, pg_temp
-AS $func$             
+AS $func$
+DECLARE
+    temp_table TEXT;         
 BEGIN
     SET LOCAL log_statement = 'none';
+    -- api user has to make sure that table_prefix is unique per session
```

Comment:
Please move this comment to the function comment. It's a critical component of the API that the caller must ensure uniqueness of the generated temp table name through the use of the prefix.